apollo-router/src/apollo_studio_interop/mod.rs

```diff
@@ -92,56 +92,66 @@ impl AggregatedExtendedReferenceStats {
-                match self
-                    .referenced_input_fields
-                    .entry(type_name.to_string())
-                    .or_default()
-                    .entry(field_name.to_string())
-                {
-                    Entry::Occupied(mut entry) => {
-                        let stats: &mut AggregatedInputObjectFieldStats = entry.get_mut();
-                        stats.referenced += if field_stats.referenced { 1 } else { 0 };
-                        stats.null_reference += if field_stats.null_reference { 1 } else { 0 };
-                        stats.undefined_reference += if field_stats.undefined_reference {
-                            1
-                        } else {
-                            0
-                        };
+                let ref_count = if field_stats.referenced { 1 } else { 0 };
+                let null_ref_count = if field_stats.null_reference { 1 } else { 0 };
+                let undefined_ref_count = if field_stats.undefined_reference {
+                    1
+                } else {
+                    0
+                };
+
+                match field_name_stats.get_mut(field_name) {
+                    Some(existing_stats) => {
+                        existing_stats.referenced += ref_count;
+                        existing_stats.null_reference += null_ref_count;
+                        existing_stats.undefined_reference += undefined_ref_count;
                     }
-                    Entry::Vacant(entry) => {
-                        entry.insert(AggregatedInputObjectFieldStats {
-                            referenced: if field_stats.referenced { 1 } else { 0 },
-                            null_reference: if field_stats.null_reference { 1 } else { 0 },
-                            undefined_reference: if field_stats.undefined_reference {
-                                1
-                            } else {
-                                0
+                    None => {
+                        field_name_stats.insert(
+                            field_name.to_string(),
+                            AggregatedInputObjectFieldStats {
+                                referenced: ref_count,
+                                null_reference: null_ref_count,
+                                undefined_reference: undefined_ref_count,
                             },
-                        });
+                        );
                     }
-                }
+                };
             }
         }
 
         // Merge enum references
         for (enum_name, enum_values) in other_stats.referenced_enums.iter() {
+            let enum_name_stats = match self.referenced_enums.get_mut(enum_name) {
+                Some(existing_stats) => existing_stats,
+                None => {
+                    self.referenced_enums
+                        .insert(enum_name.to_string(), HashMap::new());
+                    self.referenced_enums.get_mut(enum_name).unwrap()
```

Comment:
Could we change the `unwrap()` with an `expect()` please ? Could be done in a follow up PR

Reply:
I've updated these.